src/array_slice.cpp

```diff
@@ -812,4 +812,14 @@ std::vector<double> fields::get_array_metadata(const volume &where) {
 
 } // get_array_metadata
 
+std::vector<int> fields::indicate_thick_dims(const volume &where) { // indicate which dimensions have sizes larger than 1
```

Comment:
`std::vector<bool>`?

Reply:
Why is this function needed?

Reply:
I also wanted to use bool here. I changed `meep.hpp` accordingly and changed `int` to `bool` in the corresponding variable in `meep.i`. But it did not work. Perhaps I did not change `meep.i` correctly.

```
%typemap(in) int* thickness_indicator {
    $1 = (int *)thickness_indicator($input);
}
```

> Why is this function needed?

The purpose of this function is to provide assistance to the evaluation of the DFT monitor's size, which affects the size of the Jacobian `dJ`, in the `FourierFields` adjoint solver. The monitor size in a zero-thickness dimension may not be faithfully evaluated from the ivec values of the monitor's corners. To be specific, in a dimension where the thickness of the monitor is zero, the ivec values may sound like the thickness is nonzero; but in a dimension where the thickness of the monitor is nonzero, the ivec values always give reliable indications. Therefore, in order to obtain the accurate size of the DFT monitor, the code may need to know in which dimensions the monitor has zero thickness.